code/modules/mob/living/living.dm

```diff
@@ -515,8 +515,10 @@
 //Proc used to resuscitate a mob, for full_heal see fully_heal()
 /mob/living/proc/revive(full_heal = FALSE, admin_revive = FALSE)
 	SEND_SIGNAL(src, COMSIG_LIVING_REVIVE, full_heal, admin_revive)
-	if(full_heal)
+	if(full_heal && admin_revive)
 		fully_heal(admin_revive = TRUE)
+	if(full_heal)
+		fully_heal(admin_revive = FALSE)
```

Comment:
Am I missing something or wouldn't this make more sense?
```
if(full_heal)
	fully_heal(admin_revive = admin_revive)
```

Reply:
hell can't you just use fully_heal(admin_revive)?

Reply:
Good idea, I'm implementing that.